aptos-move/framework/aptos-stdlib/sources/cryptography/groups.move

```diff
@@ -0,0 +1,744 @@
+module aptos_std::groups {
+    use std::option::Option;
+
+    // Error codes
+    const E_NATIVE_FUN_NOT_AVAILABLE: u64 = 1;
+    const E_UNKNOWN_GROUP: u64 = 2;
+    const E_UNKNOWN_PAIRING: u64 = 3;
+
+    /// `BLS12_381_G1` represents a group used in BLS12-381 pairing.
+    /// The group is a prime-order group on an elliptic curve `y^2=x^3+4` defined over `Fq`.
+    /// `Fq` is a finite field with `q=0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab`.
+    /// THe order of the group `r` is 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001.
+    /// There exists a bilinear mapping from `(BLS12_381_G1, BLS12_381_G2)` to `BLS12_381_Gt`.
+    ///
+    /// A `Scalar<BLS12_381_G1>` is an integer between 0 and `r-1`.
+    ///
+    /// Function `deserialize_scalar<BLS12_381_G1>` and `serialize_scalar<BLS12_381_G1>`
+    /// assumes a 32-byte little-endian encoding of a `Scalar<BLS12_381_G1>`.
+    ///
+    /// An `Element<BLS12_381_G1>` is an element in `G1`.
+    ///
+    /// Function `serialize_element_uncompressed<BLS12_381_G1>` and `deserialize_element_uncompressed<BLS12_381_G1>`
+    /// assumes a 96-byte encoding `[b_0, ..., b_95]` of an `Element<BLS12_381_G1>`, with the following rules.
+    /// - `b_95 & 0x40` is the infinity flag.
```

Comment:
I recommend removing the term "infinity" / "point at infinity", which is specific to elliptic curve groups, and stick to "identity" / "identity element", which is the term for any abstract group.

Reply:
e.g., here you'd replace "`b_95 & 0x40` is the infinity flag" with "`b_95 & 0x40` is 1 if and only if this element is the identify element`

Reply:
Here we are describing this specific ec group, so using ec-specific terms is better?

Reply:
Mmm... great point! (No pun intended.)

Perhaps just clarify when you first use "point at infinity" that it's the "identity element of the group" (in your comments for both G1 and G2)?